### PR TITLE
Fix PlainTextBlockFilter issues

### PR DESCRIPTION
**Pull request by gonzaloUr**
These are the issues I tried to solve:
http://sonar.xwiki.org/issues/search#issues=e8de5232-8fe8-458c-8ab1-246933d4e1ff
http://sonar.xwiki.org/issues/search#issues=1b4243d4-9493-4180-bfe2-50606fd02910

### CODE REVIEW

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java [hunk 1]
```diff
@@ -78,27 +77,26 @@ public PlainTextBlockFilter(Parser plainTextParser, LinkLabelGenerator linkLabel
...
                 String label;
 
                 ResourceType resourceType = reference.getType();
-                if (ResourceType.DOCUMENT.equals(resourceType) || ResourceType.SPACE.equals(resourceType)) {
-                    label = this.linkLabelGenerator.generate(reference);
-                } else {
-                    label = reference.getReference();
-                }
+                boolean cond = ResourceType.DOCUMENT.equals(resourceType) || ResourceType.SPACE.equals(resourceType);
+                label = cond ? this.linkLabelGenerator.generate(reference) : reference.getReference();
```
Comment (tmortagne) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java]: I find this a lot less readable that the first version.

**Comment by tmortagne** [comment]
Thanks for your contribution !

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java [hunk 2]
```diff
@@ -42,15 +42,14 @@
...
-        }
-    };
+    private static final Set<Class<? extends Block>> VALID_PLAINTEXT_BLOCKS = new HashSet<Class<? extends Block>>();
+
+    static {
+        VALID_PLAINTEXT_BLOCKS.add(WordBlock.class);
+        VALID_PLAINTEXT_BLOCKS.add(SpaceBlock.class);
+        VALID_PLAINTEXT_BLOCKS.add(SpecialSymbolBlock.class);
+        VALID_PLAINTEXT_BLOCKS.add(NewLineBlock.class);
+    }
```
Comment (vmassol) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java]: BTW I'm curious. Why is this better than the previous version?

File: xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java [hunk 3]
```diff
@@ -78,27 +77,33 @@ public PlainTextBlockFilter(Parser plainTextParser, LinkLabelGenerator linkLabel
...
+        }
+
+        if (LinkBlock.class.isAssignableFrom(block.getClass()) && block.getChildren().isEmpty()) {
             ResourceReference reference = ((LinkBlock) block).getReference();
+            return filterLinkBlock(reference);
+        }
 
-            try {
-                String label;
+        return Collections.emptyList();
```
Comment (vmassol) [inline xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java]: Are you not creating a lot of exit points for the method, which would incidentally generate a new issue? :)

Comment (vmassol) [reply xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/PlainTextBlockFilter.java]: EDIT: probably not since I saw you split the method into 2.

**Pull request merged**